_posts/2019-10-16-deployment-updates.md

```diff
@@ -0,0 +1,214 @@
+
+```python
+>>> from dask.distributed import Client, SSHCluster
+>>> cluster = SSHCluster(
+...     ["host1", "host2", "host3", "host4"],
+...     connect_options={"known_hosts": None},
+...     worker_options={"nthreads": 2},
+...     scheduler_options={"port": 0, "dashboard_address": ":8797"}
+... )
+>>> client = Client(cluster)
+```
+
+This isn't what we recommend for large institutions, but it can be helpful for
+more informal groups who are just getting started.
+
+
+Dask-Jobqueue and Dask-Kubernetes Rewrite
+-----------------------------------------
+
+We've rewritten Dask-Jobqueue for SLURM/PBS/LSF/SGE cluster managers typically
+found in HPC centers and Dask-Kubernetes.  These now share a common codebase
+along with Dask SSH, and so are much more consistent and hopefully bug free.
+
+Hopefully users shouldn't notice much difference with existing workloads,
+but new features like asynchronous operation, integration with the Dask
+JupyterLab extension, and so on are more consistently available.  Also, we've
+been able to unify development and reduce our maintenance burden considerably.
+
+
+Dask-CloudProvider
+------------------
+
+For cloud deployments we generally recommend using a hosted Kubernetes or Yarn
+service, and then using Dask-Kubernetes or Dask-Yarn on top of these.
+
+However in some institutions these hosted services aren't yet accessible, and
+it's more convenient to use APIs that are more native to the particular cloud.
+The new package [Dask Cloudprovider](https://cloudprovider.dask.org) handles
+this today for Amazon's ECS API, which has been around for a long while and is
+more universally accepted.
+
+
+```python
+from dask_cloudprovider import ECSCluster
+cluster = ECSCluster(cluster_arn="arn:aws:ecs:<region>:<acctid>:cluster/<clustername>")
+
+from dask_cloudprovider import FargateCluster
+cluster = FargateCluster()
+```
+
+Dask-Gateway
+------------
+
+In some cases users may not have access to the cluster manager.  For example
+the institution may not give all of their data science users access to the Yarn
+or Kubernetes cluster.  In this the [Dask-Gateway](https://gateway.dask.org)
+project may be useful.
+It can launch and manage Dask jobs,
+and provide a proxy connection to these jobs if necessary.
+It is typically deployed with elevated permissions but managed directly by IT,
+giving them a point of greater control.
+
+<img src="https://gateway.dask.org/_images/architecture.svg" width="50%">
+
+
+GPUs and Dask-CUDA
+------------------
+
+While using Dask with multi-GPU deployments the [NVIDIA
+Rapids](https://rapids.ai) has needed the ability to specify increasingly
+complex setups of Dask workers.  They recommend the following deployment
+strategy:
+
+1.  One Dask-worker per GPU on a machine
+2.  Specify the `CUDA_VISIBLE_DEVICES` environment variable to pin that worker
+    to that GPU
+3.  If your machine has multiple network interfaces then choose the network interface closest to that GPU
```

Comment:
Maybe we should define what "closest" means in this context, which is relative to system's topology. Ideally, we would provide samples on how can users find that information, but I think this may be too out of context for this blog.